cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/replication/hbase11/LastWriteTimeObserver.java

```diff
@@ -67,8 +71,8 @@ public void postWALWrite(ObserverContext<? extends WALCoprocessorEnvironment> ct
               logKey.getTablename().toString(),
               logKey.getWriteTime(),
               logKey.getEncodedRegionName().toString());
-    if (tableUpdater != null) {
-      tableUpdater.updateTime(new String(logKey.getEncodedRegionName(), "UTF-8"), logKey.getWriteTime());
+    if (hBase11TableUpdater != null) {
+      hBase11TableUpdater.updateTime(new String(logKey.getEncodedRegionName(), "UTF-8"), logKey.getWriteTime());
```

Comment:
as mentioned before, you usually can't just create a string from a byte[] unless you know it was a string with a specific encoding to begin with. Are you sure this is safe to do? From a brief look at HRegionInfo, it looks like you should be using HRegionInfo.encodeRegionName() instead of this.

To clarify one of your comments, a byte[] cannot be used as a map key because its hash code and equals() method compare the actual byte array, not the contents of the byte array. What you use as the map key is a separate issue from how you should interpret the encoded name.